endpoints/auction.go

```diff
@@ -189,21 +189,17 @@ func (a *auction) recoverSafely(inner func(*pbs.PBSBidder, metrics.AdapterLabels
 }
 
 func (a *auction) shouldUsersync(ctx context.Context, bidder openrtb_ext.BidderName, gdprPrivacyPolicy gdprPrivacy.Policy) bool {
-	switch gdprPrivacyPolicy.Signal {
-	case "0":
-		return true
-	case "1":
-		if gdprPrivacyPolicy.Consent == "" {
-			return false
-		}
-		fallthrough
-	default:
-		if canSync, err := a.gdprPerms.HostCookiesAllowed(ctx, gdprPrivacyPolicy.Consent); !canSync || err != nil {
-			return false
-		}
-		canSync, err := a.gdprPerms.BidderSyncAllowed(ctx, bidder, gdprPrivacyPolicy.Consent)
-		return canSync && err == nil
+	gdprSignal := gdpr.SignalAmbiguous
+	// invalid data is treated as an ambiguous signal
+	if i, err := strconv.Atoi(gdprPrivacyPolicy.Signal); err == nil && (i == 0 || i == 1) {
```

Comment:
Do we need the `(i == 0 || i == 1)` check or can `gdpr.Signal(i)` take care of that logic?

Reply:
I think we need that logic. I originally intended to use `gdpr.Signal(i)` to handle other integer values as you're suggesting but it results in `gdprSignal` being set to that integer instead of one of the three signal types.

Reply:
Gotcha.

Reply:
Looking at this again, I dislike that 0 and 1 are hardcoded here far away from the context of 0 and 1 being known signals in the gdpr package. How about something like:
```
gdprSignal := gdpr.SignalAmbiguous
if i, err := strconv.Atoi(gdprPrivacyPolicy.Signal); err == nil && gdpr.IsKnownSignal(i) {
  gdprSignal = gdpr.Signal(i)
}
```
or
```
gdprSignal := gdpr.SignalAmbiguous
if signal, err := gdpr.SignalParse(gdprPrivacyPolicy.Signal); err == nil {
  gdprSignal = signal
}
```